pkg/odo/cli/component/watch.go

```diff
@@ -292,3 +298,38 @@ func NewCmdWatch(name, fullName string) *cobra.Command {
 
 	return watchCmd
 }
+
+// regenerateAdapterAndPush is used as a DevfileWatchHandler in WatchParameters; it is a wrapper around adapter.Push()
+// that first regenerates the component adapter before calling push. This ensures that it has picked up the latest
+// devfile.yaml changes
+func regenerateAdapterAndPush(pushParams common.PushParameters, watchParams watch.WatchParameters) error {
+	var adapter common.ComponentAdapter
+	adapter, err := regenerateComponentAdapterFromWatchParams(watchParams)
+	if err == nil {
+		err = adapter.Push(pushParams)
```

Comment:
Could we check for an error and if present, wrap the error with a message?

Reply:
Done.